### ISSUE TITLE
User Review

### ISSUE THREAD

**Issue by JusteenR** [OpenLiberty/draft-guide-microprofile-rest-client-async#16]
User Review:

Allow a person with a fresh mind to read the guide and execute the steps described in the guide, test whether each step is clear enough to understand, and no logic errors.

Review and walkthrough by a non-SME or a new user with a critical and fresh pair of eyes:

- [x] Record the time it takes to go through the entire guide (Around 20 min)
- [x] Test each step in the guide is clear and easy to understand
- [x] Ensure no logic errors
- [x] Resolve anything that does not work
- [x] Resolve anything usability issues (confusing or hard to follow)
- [x] Achieve basic understanding/working knowledge of the topic in the guide

When creating the JobGatewayEndpointTest class, at the Testing the gateway application step, it throws an error saying the line **package it.io.openliberty.guides.gateway;** should be **package test.java.it.io.openliberty.guides.gateway;**.

Also in the ending section Running the Tests, when I run the command `mvn verify` I get the following error:

`[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.053 s
[INFO] Finished at: 2019-08-29T10:34:20-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project gateway: Could not resolve dependencies for project io.openliberty.guides:gateway:war:1.0-SNAPSHOT: Could not find artifact io.openliberty.guides:models:jar:1.0-SNAPSHOT -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException`

**Comment by evelinec** [comment]
Java version Justeen used: 

```
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**Comment by evelinec** [comment]
The problem cannot be reproduced on my Mac nor Linux environments.

**Comment by evelinec** [comment]
The problem can be reproduced with a clean .m2 repository, run `mvn clean package` and then `mvn verify` in the child module, which is `gateway` in this guide. 

The workaround is to run `mvn clean install` in the parent directory. Then the `jar` can be found in for the `gateway` service tests. 

It comes down to the definition of the maven `package` vs `install`. `package` compiles the code and packages the code, ie, in JAR, then place the JAR in the target directory. This is the default behaviour. 

`install` also compiles and packages the code, but it also places the packaged code under the local repository, ie, .m2/repository, so other modules/projects can refer to it and grab it from the local repository. 

Thus, we need to do `mvn clean install` instead of `mvn clean package` initially.

**Comment by evelinec** [comment]
Discussed with team and agreed to changed `mvn package` to `mvn install`. Issue is resolved and the guide is updated. @JusteenR can you pls verify and close the issue when everything is good? Thank you.

**Comment by JusteenR** [comment]
When I run the `mvn clean install` command I get the following error:

-------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running it.io.openliberty.guides.system.SystemEndpointTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 51.915 s <<< FAILURE! - in it.io.openliberty.guides.system.SystemEndpointTest
[ERROR] testGetProperties(it.io.openliberty.guides.system.SystemEndpointTest)  Time elapsed: 0.007 s  <<< ERROR!
java.lang.IllegalStateException: Can not connect to Ryuk
	at it.io.openliberty.guides.system.SystemEndpointTest.<init>(SystemEndpointTest.java:68)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   SystemEndpointTest.<init>:68 » IllegalState Can not connect to Ryuk
[INFO]
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO]
[INFO] --- liberty-maven-plugin:3.0.M1:test-stop-server (test-stop-server) @ system ---
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-runtime:null:RELEASE:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2152I: Stopping server defaultServer.
[INFO] CWWKM2001I: server.config.dir is /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2001I: server.output.dir is /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2001I: Invoke command is [/Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/bin/server, stop, defaultServer].
[INFO]
[INFO] Stopping server defaultServer.
[INFO] Server defaultServer stopped.
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M1:verify (verify-results) @ system ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for reactive-demo 1.0-SNAPSHOT:
[INFO]
[INFO] reactive-demo ...................................... SUCCESS [  0.298 s]
[INFO] models ............................................. SUCCESS [  1.107 s]
[INFO] system ............................................. FAILURE [01:13 min]
[INFO] inventory .......................................... SKIPPED
[INFO] job ................................................ SKIPPED
[INFO] gateway ............................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:15 min
[INFO] Finished at: 2019-09-03T11:53:29-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-M1:verify (verify-results) on project system: There are test failures.
[ERROR]
[ERROR] Please refer to /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/test-reports/it for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :system

**Comment by JusteenR** [comment]
When I first ran `mvn clean install` I got the below errors:

When I run the mvn clean install command I get the following errors:

```
[INFO] T E S T S
[INFO] -------------------------------------------------------
[INFO] Running it.io.openliberty.guides.system.SystemEndpointTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 51.915 s <<< FAILURE! - in it.io.openliberty.guides.system.SystemEndpointTest
[ERROR] testGetProperties(it.io.openliberty.guides.system.SystemEndpointTest) Time elapsed: 0.007 s <<< ERROR!
java.lang.IllegalStateException: Can not connect to Ryuk
at it.io.openliberty.guides.system.SystemEndpointTest.(SystemEndpointTest.java:68)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR] SystemEndpointTest.:68 » IllegalState Can not connect to Ryuk
[INFO]
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
```

And...

```
[INFO] T E S T S
[INFO] -------------------------------------------------------
[INFO] Running it.io.openliberty.guides.system.SystemEndpointTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 51.915 s <<< FAILURE! - in it.io.openliberty.guides.system.SystemEndpointTest
[ERROR] testGetProperties(it.io.openliberty.guides.system.SystemEndpointTest) Time elapsed: 0.007 s <<< ERROR!
java.lang.IllegalStateException: Can not connect to Ryuk
at it.io.openliberty.guides.system.SystemEndpointTest.(SystemEndpointTest.java:68)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR] SystemEndpointTest.:68 » IllegalState Can not connect to Ryuk
[INFO]
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO]
[INFO] --- liberty-maven-plugin:3.0.M1:test-stop-server (test-stop-server) @ system ---
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-runtime:null:release:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2152I: Stopping server defaultServer.
[INFO] CWWKM2001I: server.config.dir is /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2001I: server.output.dir is /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2001I: Invoke command is [/Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/liberty/wlp/bin/server, stop, defaultServer].
[INFO]
[INFO] Stopping server defaultServer.
[INFO] Server defaultServer stopped.
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M1:verify (verify-results) @ system ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for reactive-demo 1.0-SNAPSHOT:
[INFO]
[INFO] reactive-demo ...................................... SUCCESS [ 0.298 s]
[INFO] models ............................................. SUCCESS [ 1.107 s]
[INFO] system ............................................. FAILURE [01:13 min]
[INFO] inventory .......................................... SKIPPED
[INFO] job ................................................ SKIPPED
[INFO] gateway ............................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:15 min
[INFO] Finished at: 2019-09-03T11:53:29-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-M1:verify (verify-results) on project system: There are test failures.
[ERROR]
[ERROR] Please refer to /Users/Justeen.Randev@ibm.com/Desktop/draft-guide-microprofile-rest-client-async/start/system/target/test-reports/it for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR] mvn -rf :system
```

When I re-ran the command it was successful. 
The api does not open on the safari browser but it does open on the chrome browser. I am unable to run the curl command to POST I get the error:

`curl: (6) Could not resolve host: localhost`

**Comment by evelinec** [comment]
When I re-ran the command it was successful.
=> hm, will look into this and see if I can reproduce this issue. 

The api does not open on the safari browser but it does open on the chrome browser. 
=> Will investigate. 

I am unable to run the curl command to POST I get the error: curl: (6) Could not resolve host: localhost
=> This is the timing issue. You need to wait for the services to become available. This can be up to 5 minutes. Let me know and if you can record the time it becomes available that'd be helpful. Thank you.

**Comment by evelinec** [comment]
Since the issue encountered is environmental on the Justeen's laptop, team decided to ask Justeen to run the guide's test on another system to complete this user review and testing task.

**Comment by evelinec** [comment]
@JusteenR tested the guide successfully on a Linux VM. She agreed to close this issue.
